src/main/java/core/basesyntax/strategy/impl/PurchaseHandler.java

```
@@ -0,0 +1,14 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.FruitStorage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class PurchaseHandler implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        int newAmount = transaction.getQuantity()
+                - FruitStorage.calculationStorage.get(transaction.getFruit());
+        FruitStorage.calculationStorage.put(transaction.getFruit(), newAmount);

```

let's throw an exception if there is not enough fruit in storage